ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -10633,6 +10643,16 @@ ABoundActivity * HqlCppTranslator::doBuildActivityOutput(BuildCtx & ctx, IHqlExp
         addFilenameConstructorParameter(*instance, "getFileName", filename);
     }
 
+    IHqlExpression * signedAttr = expr->queryAttribute(_fileAccessSigned_Atom);
```

Comment:
this code is repeated more than 2 times it should be commoned up into a single functions.